snapcraft/parts/plugins/initrd_build.sh

```diff
@@ -0,0 +1,503 @@
+#!/bin/sh
+
+set -eux
+
+# clean if we fail
+trap 'clean "$chroot_home"' EXIT INT
+
+# Set initrd root
+initrd_root="${CRAFT_PART_SRC}/uc-initramfs-build-root"
+
+# snappy-dev PPA fingerprint for ubuntu-core-initramfs deb
+ppa_fingerprint=F1831DDAFC42E99D
+
+# Get the kernel version
+kver="$(basename "${CRAFT_STAGE}/modules/"*)"
+kernel_modules="${CRAFT_STAGE}/modules"
+kernel_firmware="${CRAFT_STAGE}/firmware"
+
+# Simple tracker for chroot state
+base_created="${CRAFT_PART_SRC}/.base_created"
+base_configured="${CRAFT_PART_SRC}/.base_configured"
+
+# Args passed from initrd.py
+# --build-efi-image|--efi-image-{key,cert} are from snapcraft/parts/plugins/v2/initrd.py
+for arg; do
+  case $arg in
+    modules=*)        rd_mods=${arg#*=}  ;; # valid: list, for mod.ko pass mod
+    firmware=*)       rd_fw=${arg#*=}    ;; # valid: list, relative path to "${CRAFT_STAGE}"
+    addons=*)         rd_ao=${arg#*=}    ;; # valid: list, relative path to "${CRAFT_STAGE}"
+    overlay=*)        rd_ol=${arg#*=}    ;; # valid: list, relative path to "${CRAFT_STAGE}"
+    efi-image=*)      efi_img=${arg#*=}  ;; # valid: True|False
+    efi-image-key=*)  img_key=${arg#*=}  ;; # valid: path/to/key.key
+    efi-image-cert=*) img_cert=${arg#*=} ;; # valid: path/to/cert.pem
+  esac
+done
+
+# Simple mount wrapper
+mnt() {
+  dest=$1
+  shift
+  mountpoint "$dest" || mount "$@" "$dest"
+}
+
+# Simple umount wrapper
+umnt() {
+  dir=$1
+  shift
+  mountpoint "$dir" || umount "$@" "$dir"
+}
+
+# clean any existing mounts for the chroot function
+clean() {
+  chroot_home="$1"
+
+  if [ -z "$chroot_home" ]; then
+    echo "No chroot to clean"
+    return
+  fi
+
+  # ensure no chroot processes are left running
+  for pid in "/proc/"*; do
+    if [ -e "${pid}/root" ] && [ "$(readlink -f "${pid}/root")" = "$chroot_home" ]; then
+      echo "Killing PID $pid inside $chroot_home chroot"
+      kill -9 "$pid"
```

Comment:
```suggestion
      kill -9 "$pid" || continue
```

Some processes will be *unkillable*. We shouldn't fail for that alone; that's why we lazy unmount.